responsibleai/responsibleai/_managers/counterfactual_manager.py

```diff
@@ -329,8 +365,88 @@ def name(self):
+            with open(config_path, 'w') as config_file:
+                json.dump(
+                    counterfactual_config.get_config_as_dict(),
+                    config_file)
+
+            result_path = counterfactual_state_dir / 'result.json'
+            with open(result_path, 'w') as result_file:
+                json.dump(
+                    counterfactual_config.get_result(),
+                    result_file)
+
+    @classmethod
+    def _load(cls, path, model_analysis):
+        """Load the CounterfactualManager from the given path.
+
+        :param path: The directory path to load the CounterfactualManager from.
+        :type path: str
+        :param model_analysis: The loaded parent ModelAnalysis.
+        :type model_analysis: ModelAnalysis
+        """
+        this = cls.__new__(cls)
+        counterfactual_dir = Path(path)
+
+        # Rehydrate model analysis data
+        this.__dict__['_model'] = model_analysis.model
+        this.__dict__['_train'] = model_analysis.train
+        this.__dict__['_test'] = model_analysis.test
+        this.__dict__['_target_column'] = model_analysis.target_column
+        this.__dict__['_task_type'] = model_analysis.task_type
+        this.__dict__['_categorical_features'] = \
+            model_analysis.categorical_features
+
+        this.__dict__['_counterfactual_config_list'] = []
+        import os
+        all_cf_dirs = os.listdir(counterfactual_dir)
+        for counterfactual_config_dir in all_cf_dirs:
+            config_path = \
+                Path(path) / counterfactual_config_dir / 'config.json'
+
+            with open(config_path, 'r') as config_file:
+                cf_config = json.load(config_file)
+
+            counterfactual_config = CounterfactualConfig(
+                method=cf_config['method'],
+                continuous_features=cf_config['continuous_features'],
+                total_CFs=cf_config['total_CFs'],
+                desired_class=cf_config['desired_class'],
+                desired_range=cf_config['desired_range'],
+                permitted_range=cf_config['permitted_range'],
+                features_to_vary=cf_config['features_to_vary'],
+                feature_importance=cf_config['feature_importance'])
+
+            result_path = \
+                Path(path) / counterfactual_config_dir / 'result.json'
+
+            with open(result_path, 'r') as result_file:
+                cf_result = json.load(result_file)
+
+            if cf_result['counterfactual_obj'] is not None:
+                counterfactual_config.counterfactual_obj = \
+                    CounterfactualExplanations.from_json(
+                        cf_result['counterfactual_obj'])
+            else:
+                counterfactual_config.counterfactual_obj = None
+            counterfactual_config.has_computation_failed = \
+                cf_result['has_computation_failed']
+            counterfactual_config.failure_reason = cf_result['failure_reason']
```

Comment:
nit: maybe move some of these repeating strings to be constants at the top of this file

Reply:
Done